src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java

```diff
@@ -262,6 +267,26 @@ public InnReachResponseDTO returnUncirculated(String trackingId, String centralC
     }
   }
 
+  @Override
+  public InnReachResponseDTO borrowerRenew(String trackingId, String centralCode, BorrowerRenewDTO borrowerRenew) {
+    var transaction = getTransaction(trackingId, centralCode);
+    var transactionState = transaction.getState();
+    var calculatedDueDate = transaction.getHold().getNeedBefore();
+    var requestedDueDate = borrowerRenew.getDueDateTime();
+
+    if (calculatedDueDate >= requestedDueDate) {
+      transaction.setState(BORROWER_RENEW);
+      transactionRepository.save(transaction);
+    } else if (transactionState == ITEM_SHIPPED) {
+      transaction.setState(RECALL);
+      transactionRepository.save(transaction);
+      } else {
+      throw new IllegalArgumentException(UNEXPECTED_TRANSACTION_STATE + transactionState.name());
+    }
+
+    return success();
+  }
```

Comment:
If I understand the requirements correctly we need to make a call to FOLIO circulation API to

> Attempt to renew the loan associated with the Item Hold transaction

Reply:
Thanks for help. Fixed